### PR TITLE
Proposed all_bookmarks() function for Pyto bookmarks module

### PR DESCRIPTION
**Pull request by jx256**
Hi Emma (@ColdGrub1384),

I created a utility function called `all_bookmarks()` for the Pyto bookmarks module, which I have found useful. Below is it’s description.
 
Best regards,
Joe Gering (@jx256)
============================
Function to return a list of dictionaries 
for all bookmarks with name, path, and 
type (file or folder) keys.

### CODE REVIEW

File: Lib/bookmarks.py [hunk 1]
```diff
@@ -148,3 +148,37 @@ class FolderBookmark(StoredBookmark):
     """
 
     __type__ = "public.folder"
+
+
+def AllBookmarks():
```
Comment (cclauss) [inline Lib/bookmarks.py:153]:
```suggestion
def all_bookmarks():
```
In Python, classes are CamelCase while functions and methods are snake_case.

File: Lib/bookmarks.py [hunk 2]
```diff
@@ -148,3 +148,37 @@ class FolderBookmark(StoredBookmark):
...
+    """
+    Function to return a list of dictionaries 
+    for all bookmarks with name, path, and 
+    type (file or folder) keys.
+    JG 2021-06-11
+    """
+    answer = []
+    bkmks = __stored_bookmarks__()
+    for name in bkmks.keys():
+        value = bkmks[name]
```
Comment (cclauss) [inline Lib/bookmarks.py:163]:
```suggestion
    for name, value in bkmks.items():
```

Comment (cclauss) [inline Lib/bookmarks.py:163]:
```suggestion
    for name, value in __stored_bookmarks__().items():
```

File: Lib/bookmarks.py [hunk 3]
```diff
@@ -148,3 +148,37 @@ class FolderBookmark(StoredBookmark):
...
+        data = NSData.alloc().initWithBase64EncodedString(value, options=0)
+        url = NSURL.URLByResolvingBookmarkData(
+            data,
```
Comment (cclauss) [inline Lib/bookmarks.py:166]:
```suggestion
        url = NSURL.URLByResolvingBookmarkData(
            NSData.alloc().initWithBase64EncodedString(value, options=0),
```

File: Lib/bookmarks.py [hunk 4]
```diff
@@ -148,3 +148,37 @@ class FolderBookmark(StoredBookmark):
...
+            relativeToURL=None,
+            bookmarkDataIsStale=None,
+            error=None,
+            )
+        path = str(url.path)
+        typ = "unknown"
+        if (os.path.isfile(path)):
+            typ = "file"
+        if (os.path.isdir(path)):
+            typ = "folder"
```
Comment (cclauss) [inline Lib/bookmarks.py:177]:
```suggestion
        if os.path.isfile(path):
            typ = "file"
        else:
            typ = "folder" if os.path.isdir(path) else "unknown"
```

File: Lib/bookmarks.py (hunk 1 above)
Comment (jx256) [reply Lib/bookmarks.py]: Thanks for the update. I wasn't aware of that convention. Ironically, in an earlier working version, I called the function all_bookmarks.

**Review by cclauss — APPROVED** [review]
LGTM

**Comment by jx256** [comment]
@cclauss, I did “one last commit” on this branch after your approval.

**Pull request closed without merging**